### PR TITLE
add Fill In The Blanks: Closures

### PR DESCRIPTION
**Pull request by Lotterleben**

### CODE REVIEW

File: fill_in_the_blanks/closures.adoc [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    // 👀  a common use case for closures is to transform collections
+    //     using e.g. `map()` and `filter()`.
+
+    // ✅ TODO: use `map()` to increase the bale size with every "hay" in the haystack
+    let mut bale = BaleOfStraw{ size: 0 };
+
+    let new_haystack: Vec<_> = haystack
+        .into_iter()
+        .filter(|element | *element == "hay")
+        .map( /* increment bale size here */ )
```
Comment (spookyvision) [inline fill_in_the_blanks/closures.adoc:87]: if the *only* intent is to increment bale size, change `map` to `inspect`

Comment (Lotterleben) [reply fill_in_the_blanks/closures.adoc:87]: the intent is to introduce people to map :D how about we change this to...
```suggestion
        .map( /* remove straws and increment bale size here */ )
```
?

Comment (spookyvision) [reply fill_in_the_blanks/closures.adoc:87]: what code do you have in mind for "remove straws"?

Comment (Lotterleben) [reply fill_in_the_blanks/closures.adoc:87]: I was looking for something like:

```
    let new_haystack: Vec<_> = haystack
        .into_iter()
        .filter(|element| *element == "hay")
        .map(|element| bale.size += 1)
        .collect();
```

I'll also add an example solution

File: assignments/solutions/fill_in_the_blanks/src/closures.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    // 👀  a common use case for closures is to transform collections
+    //     using e.g. `map()` and `filter()`.
+
+    // ✅ TODO: use `map()` to increase the bale size with every "hay" in the haystack
+    let mut bale = BaleOfStraw { size: 0 };
+
+    let empty_haystack: Vec<_> = haystack
+        .into_iter()
+        .filter(|element| *element == "hay")
+        .map(|_| bale.size += 1)
```
Comment (spookyvision) [inline assignments/solutions/fill_in_the_blanks/src/closures.rs:50]: how about this use of `map`?

```rust
    let haystack = vec!["hay", "hay", "hay", "needle", "hay", "hay"];
    // ...
    let heystack: Vec<_> = haystack
        .into_iter()
        .filter(|element| *element == "hay")
        .map(|item| item.replace("hay", "hey"))
        .collect();
    println!("hey! {:?}", heystack);
```

Comment (Lotterleben) [reply assignments/solutions/fill_in_the_blanks/src/closures.rs:50]: haha, I like the play on words a lot! Unfortunately I think the difference between "hay" and "hey" is easy to miss during the frenzy of a training so I will sliiightly modify it though

**Pull request merged**